server/src/main/java/org/apache/druid/client/coordinator/CoordinatorClient.java

```diff
@@ -121,4 +127,35 @@ public Boolean isHandOffComplete(String dataSource, SegmentDescriptor descriptor
       throw new RuntimeException(e);
     }
   }
+
+  public List<DataSegment> getDatabaseSegmentDataSourceSegments(String dataSource, List<Interval> intervals)
+  {
+    try {
+      FullResponseHolder response = druidLeaderClient.go(
```

Comment:
There's a difference between using `SegmentListUsedAction` and directly calling this API: users can configure the retry policy on failure for all remote task actions including `SegmentListUsedAction`. (I'm not sure this is intended, but the actual max number of retries for remote task actions is `RetryPolicy.maxNumRetries * DruidLeaderClient.MAX_RETRIES`).

Can we do the same thing for the coordinator client to provide the same experience?

Reply:
OK, what are you proposing exactly?

(a) Should there be a separate retry policy config bound elsewhere, or should this reuse `druid.peon.taskActionClient.retry`
(b) Should the retry be performed inside this method here, or in the caller?

Reply:
(a) I think we should reuse the same configuration name for backward compatibility.
(b) I think it's fine to retry in this method.

Reply:
Hmm, the whole RetryPolicy set of classes is defined in `indexing-service`, so I think I'll ignore your suggestion on (b) and do it in IngestSegmentFirehoseFactory.

Reply:
I'm fine by moving `CoordinatorClient` to `indexing-service` along with `CoordinatorBasedSegmentHandoffNotifier`, `CoordinatorBasedSegmentHandoffNotifierConfig`, and `CoordinatorBasedSegmentHandoffNotifierFactory` , but it's up to you.

Reply:
`CoordinatorBasedSegmentHandoffNotifierFactory` is used by RealtimeModule which seems to otherwise use very little (though nonzero) from `indexing-service`. I think doing this move feels a little intense to me and I'd like to leave it alone if that's OK.